server/server.go

```diff
@@ -192,6 +200,13 @@ func (s *Server) WithCompilerErrorLimit(limit int) *Server {
 	return s
 }
 
+// WithDiagnosticsBufferSize sets the size on the diagnostics buffer stored by the
+// server.
+func (s *Server) WithDiagnosticsBufferSize(size int) *Server {
```

Comment:
Did you consider making the diagnostics buffer itself a parameter for the server? Size seems like an implementation detail of the current diagnostics buffer--instead the diagnostics buffer could be passed as an interface.